### PR TITLE
Remove numpy dependency

### PR DESCRIPTION
**Pull request by maresb**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [X] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by maresb** [comment]
@conda-forge-admin, please rerender

**Comment by maresb** [comment]
@ocefpaf, I don't see numpy as an upstream dependency. Not sure if there's some obscure reason why it's actually required? Let's see what happens when I try to remove it. (It was added in an "update with Grayskull" commit.)

**Comment by maresb** [comment]
@ocefpaf, it's green :rocket:

File: recipe/meta.yaml [hunk 1]
```diff
@@ -23,7 +23,6 @@ requirements:
     - python >=3.8
     - click >=7.0
     - rasterio >=1.3.3
-    - numpy >=1.15,<2.dev0
```
Comment (ocefpaf) [inline recipe/meta.yaml:26]: Rasterio will bring numpy. The pin here was probably due to lack of ,<2 there. (Not sure.) Anyway, the issue is gone and that was not the best way to fix this.

Thanks for correcting it.

Comment (maresb) [reply recipe/meta.yaml:26]: Thanks so much @ocefpaf!!!

Now I can install rio-cogeo alongside Numpy 2. :tada:

**Pull request merged**